### PR TITLE
PSR-11 Container - Factory

### PR DESCRIPTION
**Pull request by imonteiro**
Add factory, using PSR-11 standard.

@kokspflanze just tested with `laminas-servicemanager` and works as expected.

### CODE REVIEW

File: src/Factory/AlphaVantageFactory.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        $config      = $container->has('config') ? $container->get('config') : [];
+        $options     = $config['alpha_vantage'] ?? [];
+        if (! isset($options['api_key'])) {
+            throw new UnexpectedValueException("Missing 'api_key' key for the alpha vantage configuration");
+        }
+
+        $option = new Options();
+        $option->setApiKey($options['api_key']);
+
+        $client = new Client($option);
```
Comment (kokspflanze) [inline src/Factory/AlphaVantageFactory.php]: why you added a variable for the instance of the client?

i would just write `return new Client($option);`

**Review by kokspflanze — CHANGES_REQUESTED** [review]

File: src/Factory/AlphaVantageFactory.php (hunk 1 above)
Comment (imonteiro) [reply src/Factory/AlphaVantageFactory.php]: Fixed

**Comment by imonteiro** [comment]
Travis is failing: `Could not authenticate against github.com`. Maybe to much calls (https://developer.github.com/v3/#rate-limiting).

I added PHPCS. Should be a different PR?

File: composer.lock [hunk 2]
```diff
@@ -1763,5 +1815,8 @@
     "platform": {
         "php": "^7.1"
     },
-    "platform-dev": []
+    "platform-dev": [],
+    "platform-overrides": {
+        "php": "7.1.0"
```
Comment (kokspflanze) [inline composer.lock]: did you need help with php 71?
overrides is not so good for the testing on different versions

**Review by kokspflanze — CHANGES_REQUESTED** [review]
no is okay, in 1 pr

but can check the composer.lock?

i can fix this too, if you need help

File: composer.lock (hunk 2 above)
Comment (imonteiro) [reply composer.lock]: I agree. Locally I have PHP 7.4 and forced on composer global PHP 7.1.0.

If you can help I will appreciate. Which is the best approach to generate `composer.lock` without "writting" `platform.php` on it? Should be removed mannually this entry?

**Pull request merged**